zebra-state/src/service/finalized_state/disk_format/shielded.rs

```diff
@@ -140,3 +152,49 @@ impl FromDisk for orchard::tree::NoteCommitmentTree {
             .expect("deserialization format should match the serialization format used by IntoDisk")
     }
 }
+
+impl IntoDisk for sapling::tree::Node {
+    type Bytes = Vec<u8>;
+
+    fn as_bytes(&self) -> Self::Bytes {
+        self.as_ref().to_vec()
+    }
+}
+
+impl IntoDisk for orchard::tree::Node {
+    type Bytes = Vec<u8>;
+
+    fn as_bytes(&self) -> Self::Bytes {
+        self.to_repr().to_vec()
+    }
+}
+
+impl<Node: IntoDisk<Bytes = Vec<u8>>> IntoDisk for NoteCommitmentSubtree<Node> {
+    type Bytes = Vec<u8>;
+
+    fn as_bytes(&self) -> Self::Bytes {
+        [self.end.as_bytes().to_vec(), self.node.as_bytes()].concat()
+    }
+}
+
+impl FromDisk for sapling::tree::Node {
+    fn from_bytes(bytes: impl AsRef<[u8]>) -> Self {
+        Self::from_repr_unchecked(bytes.as_ref())
+    }
+}
+
+impl FromDisk for orchard::tree::Node {
+    fn from_bytes(bytes: impl AsRef<[u8]>) -> Self {
+        Self::from_repr_unchecked(bytes.as_ref())
+    }
+}
+
+impl<Node: FromDisk> FromDisk for NoteCommitmentSubtreeData<Node> {
+    fn from_bytes(disk_bytes: impl AsRef<[u8]>) -> Self {
+        let bytes = disk_bytes.as_ref();
+        Self::new(
+            Height::from_bytes(&bytes[..3]),
+            Node::from_bytes(&bytes[3..35]),
```

Comment:
Is there a way to avoid using integer constants here?
It might help to do it like this:
https://github.com/ZcashFoundation/zebra/blob/7fe040ed8ebb2421a920702c98a00dc5beb51953/zebra-state/src/service/finalized_state/disk_format/block.rs#L290